readme.md

```diff
@@ -334,6 +334,21 @@ got('todomvc.com', {
 });
 ```
 
+If you require different agents for different protocols, you can pass in all of them through the `agents` option. This is necessary because a request to one protocol might redirect to another. In such a scenario, `got` will switch over to the right protocol agent for you.
+
+```js
+const got = require('got');
+const HttpAgent = require('agentkeepalive')
+const HttpsAgent = HttpAgent.HttpsAgent
+
+got('todomvc.com', {
+	agents: {
```

Comment:
Better to just overload the `agent` option. It's confusing to have two so similarly named options.